main.go

```diff
@@ -218,7 +221,24 @@ func runMany(dbpath string, getPort func() int, many, bucketSize, bsCon int, rel
 	start := time.Now()
 	var hosts []host.Host
 	var dhts []*dht.IpfsDHT
-	uniqpeers := make(map[peer.ID]struct{})
+
+	var hyperLock sync.Mutex
+	hyperlog := hyperloglog.New()
+	var peersConnected int64
+
+	notifiee := &network.NotifyBundle{
+		ConnectedF: func(_ network.Network, v network.Conn) {
+			hyperLock.Lock()
+			hyperlog.Insert([]byte(v.RemotePeer()))
```

Comment:
Uber-nit: we can reuse a scratch buffer and copy into it to avoid an allocation. Almost certainly not worth it...

Reply:
I think sync would add enough overhead for it not to be worth it.